src/mask.rs

```diff
@@ -266,17 +280,43 @@ mod imp {
     }
 }
 
+/// (Un-)masks input bytes with the framing key, one byte at once.
+///
+/// See [`fallback_frame`] for more details.
+fn one_byte_at_once(key: [u8; 4], input: &mut [u8], offset: usize) {
+    for (index, byte) in input.iter_mut().enumerate() {
+        *byte ^= key[(index + offset) % key.len()];
+    }
+}
+
 /// (Un-)masks input bytes with the framing key.
 ///
 /// The input bytes may be further in the payload and therefore the offset into
 /// the payload must be specified.
 ///
 /// This is used as the internal implementation in non-SIMD builds and as a
 /// fallback in SIMD builds.
-pub fn fallback_frame(key: [u8; 4], input: &mut [u8], offset: usize) {
-    for (index, byte) in input.iter_mut().enumerate() {
-        *byte ^= key[(index + offset) % key.len()];
+pub fn fallback_frame(key: [u8; 4], input: &mut [u8], mut offset: usize) {
+    let (prefix, aligned_data, suffix) = unsafe { input.align_to_mut::<i64>() };
```

Comment:
Could this slow down simd mask implementations when `input` is unaligned? Should they not call `one_byte_at_once` instead?

Reply:
I don't really think so. E.g. AVX512 could have up to 63 bytes of unaligned data, which can usually be masked in ~1/4 of the cycles using this implementation.

Reply:
I'm tempted to say we don't even need to use `align_to_mut` here, sincec we can just convert 8x u8 into an i64 no matter whether they are aligned. That would probably make it easier to reason about this.

Reply:
I forgot to mention, but shouldn't the fallback unmask implementation also call `one_byte_at_once`? It's odd to call auto vectorizing code when SIMD is not requested. There's module level documentation about this too:

https://github.com/Gelbpunkt/tokio-websockets/blob/81e8f4319916a11bde2d4f5b7131fbbb63d60482/src/mask.rs#L11-L13

(For the record I think `simd` flags are an anti-pattern)

> I'm tempted to say we don't even need to use `align_to_mut` here, since we can just convert 8x u8 into an i64 no matter whether they are aligned. That would probably make it easier to reason about this.

But the input length varies on the lane width.

Reply:
> (For the record I think `simd` flags are an anti-pattern, so I'm totally fine with changing this behavior)

I actually agree. I think SIMD should be enabled when the necessary CPU target flags are set at build time, not when a crate feature is enabled. IIRC we unconditionally enable the feature in twilight. I'm down for removing it.

Reply:
The latest changes move from a `simd` feature to runtime feature detection.